cpp/react-native-hashing.cpp

```diff
@@ -23,8 +24,26 @@ namespace RNHashing {
             return jsi::Value(rt, jsi::String::createFromUtf8(rt, converter(input)));
         }
                         );
+        auto md5 = jsi::Function::createFromHostFunction(rt,
+                                                         jsi::PropNameID::forAscii(rt, "md5"),
+                                                         1,
+                                                         [](jsi::Runtime &rt, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value {
+            if (count == 0 || count > 1) {
+                jsi::detail::throwJSError(rt, "md5 requires one arguement");
+                return {};
+            }
+            if (!args[0].isString()) {
+                jsi::detail::throwJSError(rt, "arguement for md5 needs to be of type string");
+                return {};
+            }
+            string input = args[0].asString(rt).utf8(rt);
+            MD5 converter;
```

Comment:
In c++ every bit matters.

You are creating `MD5 converter;` on stack every time you invoke the function, you can potentially reuse it by moving it out of the function.



don't forget to capture(https://www.learncpp.com/cpp-tutorial/lambda-captures/) mdConverter if you move it ouside lambda function. 
you can do
```
 const MD5 md5Converter;
 
 auto md5 = jsi::Function::createFromHostFunction(rt,
                                                         jsi::PropNameID::forAscii(rt, "md5"),
                                                         1,
                                                         [& md5Converter](jsi::Runtime &rt, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value {
                                                         
                                                         ```

Reply:
Agree

Reply:
Fixed under ca7d592